content/blog/infrastructure-as-code-on-hpe-greenlake-using-terraform-–-part-2.md

```diff
@@ -0,0 +1,363 @@
+
+```
+# Power state     
+power = "poweron"
+
+```
+
+While *poweron* is the only valid option when creating a new *resourcetype hpegl_vmaas_instance*, other values are available for lifecycle management, such as *poweroff* and *suspend*.
+
+Edit the Terraform configuration file and change the power desired state to be *poweroff*.
+
+```
+# Power state         
+power = "poweroff"
+
+```
+
+Save and run a **terraform apply** command, which will prompt you to accept the following change:
+
+```markdown
+Terraform will perform the following actions:
+
+# hpegl_vmaas_instance.DidierTest1 will be updated in-place
+  ~ resource "hpegl_vmaas_instance" "DidierTest1" {
+        id                 = "149"
+        name               = "DidierTest1"
+      ~ power              = "poweron" -> "poweroff"
+```
+
+Pretty soon afterwards, you can check out the HPE GreenLake console and see that the VM status was changed to stopped.
+
+![VM is now stopped](/img/terraform-greenlake-part2-blog-picture2.png "VM is now stopped")
+
+### Use case 2: Setting up tags and labels
+
+That was straight forward, right?, Now, restart the VM and try something else. You might want to test out tabs and labels. As organizations scale their cloud environments, they often need to define methodologies for organizing resources. For this, they can leverage tags and labels. Tags consist of key/value pairs that make it easier to search for, or filter, your cloud resources based on categories relevant to the organization. Another option is to attach labels, which are simple values, to your VMs in order to keep track of what it’s used for or who it belongs to.
+
+Why don’t you try adding metadata to the VM using tags and labels. According to the [documentation](https://github.com/HPE/terraform-provider-hpegl/blob/main/docs/resources/vmaas_instance.md), you can add labels using the following syntax:
+
+```markdown
+# Using labels
+labels = ["hackshack", "hpedev"]
+
+```
+
+And you can add tags with the following syntax:
+
+```markdown
+# Using tags
+tags = {
+        team  = "HPE Developer"
+        support = "gold"
+  }
+```
+
+Save and apply those changes with **terraform apply**, wait a little and look at the VM details. You can see the labels and the tags in the capture below:
+
+![tags and labels applied to VM](/img/terraform-greenlake-part2-blog-picture3.png "tags and labels applied to VM")
+
+### Use case 3: Get me more disk space
+
+Another typical use case would be to add another disk to a VM, say a data volume for application usage. The syntax for this is the same as you used to create the VM with its *root_vol*, already visible in the Storage details of the VM:
+
+![VM was created with one disk: root_vol](/img/terraform-greenlake-part2-blog-picture4.png "VM was created with one disk: root_vol")
+
+Go ahead and add the following code snippet to your configuration file:
```

Comment:
I like the way you tell to append the config file ... this could be confusing for others use cases